mesonbuild/backend/ninjabackend.py

```diff
@@ -2381,13 +2382,13 @@ def generate_dynamic_link_rules(self):
 
                 options = self._rsp_options(compiler)
                 self.add_rule(NinjaRule(rule, command, args, description, **options, extra=pool))
-            if self.environment.machines[for_machine].is_aix():
-                rule = 'AIX_LINKER{}'.format(self.get_rule_suffix(for_machine))
-                description = 'Archiving AIX shared library'
-                cmdlist = compiler.get_command_to_archive_shlib()
-                args = []
-                options = {}
-                self.add_rule(NinjaRule(rule, cmdlist, args, description, **options, extra=None))
+        if self.environment.machines[for_machine].is_aix():
+            rule = 'AIX_LINKER{}'.format(self.get_rule_suffix(for_machine))
+            description = 'Archiving AIX shared library'
+            cmdlist = compiler.get_command_to_archive_shlib()
+            args = []
+            options = {}
+            self.add_rule(NinjaRule(rule, cmdlist, args, description, **options, extra=None))
```

Comment:
Why is this indentation being removed here? It looks like you are dropping off the end of the for loop and only producing one rule, using whichever for_machine turns out to be last...

Reply:
@eli-schwartz 

You are right here. I changed it back. Also, while I changed it back, I saw an error while building Scipy, which looked like this:

```
File "/meson_install/opt/freeware/lib/python3.9/site-packages/mesonbuild/backend/ninjabackend.py", line 1315, in generate_rules
    self.generate_dynamic_link_rules()
  File "/meson_install/opt/freeware/lib/python3.9/site-packages/mesonbuild/backend/ninjabackend.py", line 2388, in generate_dynamic_link_rules
    cmdlist = compiler.get_command_to_archive_shlib()
  File "/meson_install/opt/freeware/lib/python3.9/site-packages/mesonbuild/compilers/compilers.py", line 963, in get_command_to_archive_shlib
    return self.linker.get_command_to_archive_shlib()
AttributeError: 'NoneType' object has no attribute 'get_command_to_archive_shlib'
```
I added a guard in the compilers.py file. Kindly see the changes there and let me know. This is because linker object was none.